### ISSUE TITLE
Save a "standard" (non R-package) project on WINDOWS network  

### ISSUE THREAD

**Issue by gwd666** [rstudio/rstudio#7716]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.1073, 1.2.5033 
    OS Version      :  MS Windows 10 Enterprise - Version 10.0.17763 Build 17763
    R Version       : R-3.6.3, R-3.5.3 

### Steps to reproduce the problem

- Open RStudio 
- Click File -> New Project 
In New Project Wizard: 
- Create Project Select 'New Directory' 
- Project Type: Select 'New Project' 
![image](https://user-images.githubusercontent.com/57595487/92093041-23a32c00-edd3-11ea-9b4b-1ec24d29c8c9.png)
(Alternatively selecting "Open in New Session" in this window has the same result) 
- Click 'Create Project' 
RESULT 
![image](https://user-images.githubusercontent.com/57595487/92093291-77157a00-edd3-11ea-80bc-fa581e6a78a0.png)
After clicking OK on the two error messages 
![image](https://user-images.githubusercontent.com/57595487/92093780-0589fb80-edd4-11ea-82b5-abb3078340df.png)

### Describe the problem in detail
See above (above screenshot) regarding ERROR messages.  

**If you select Project type: R package this error does not occur.**
 
Error also happens if I Create Project from 'Existing Directory' when selecting a directory on the P: (Network) Drive. 
  
Error is tricky to reproduce because in "some" network folders the project will manage to get created. 
As far as I have identified the difference is that this "P" drive is a form of "GLOBAL" share - meaning that it is shared on some form of "highest level" whereas the network shares that allow to create 'New Projects' have an additional layer of hierarchy in the sharing structure eg there is a network drive (call it U:) where every user will have his user folder to which he then has access rights, on these one-level elevated access structure, so to speak, projection creation succeeds.   

CURRENT WORKAROUND 
Create 'New Project' type project locally or on one of the drives where it works and copy/paste to the (GLOBAL shared) network drive. 
Then click 'Open Project' and browse to that pasted directory (on GLOBAL) and open the project there. 

### Describe the behavior you expected
Creation of projects of 'New Project' type on shared network drive.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report. It looks like the issue is also described here:

https://stackoverflow.com/questions/44393241/boostfilesystemrelative-cannot-access-the-file-because-it-is-being-used-by

Fortunately, it should be fixed in the next Boost release:

https://github.com/boostorg/filesystem/commit/726d2bfa35b42144986cdd964f49374204d816bc

so hopefully once we take our next Boost update this issue will be resolved.

**Comment by gwd666** [comment]
@kevinushey: thank you for the quick info 
Will close this as soon as I install and test with this next update - keeping it open to as a reminder. 
Final question will this boost update also effect RStudio 1.2.5033? Thanks

**Comment by gwd666** [comment]
@kevinushey 
I have come to notice another issue in connection with (those) Windows network paths
When opening existing projects eg from one a where the above BUG strikes you with not being able to SAVE standard projects and then when switching (either directly via the Project dropdown or via 'close' and 'open' another existing) project it is really good advise to NOT attempt that on two different drive letters - it takes AGES (approx 5-10 minutes) to complete any of these transactions (eg closing a project, which by the way completely hangs itself most of the times). 
AND heaven forbid you attempt to quit RStudio during that (Closing project) process - I was only able to make RStudio start up and not hang itself during that startup again AFTER deleting the (whole) \LOCALAPPDATA\RStudio-Desktop folder. This frequently happened when attempting 'Session Quit' during 'Closing project' and then killing RStudio via the WINdows taskmanager after a couple approx. 15 minutes). 
This basically makes dev work impossible as soon as there I have projects one more than one drive (letter).

**Comment by kevinushey** [comment]
> Final 2 questions when do u expect yr update and will this boost update also effect RStudio 1.2.5033? Thanks

Sorry, but I'm not sure -- most likely, this update won't happen until next year; perhaps with the next version of RStudio (v1.5). I'm realize this isn't what you wanted to hear, but updating dependencies is time-consuming as it takes time for the team to run down regressions and other issues that typically arise after an update.

Per your most recent comment -- this sounds extraordinarily frustrating; I'm sorry that RStudio has been causing you so much trouble here. :-/ Are you able to collect any diagnostic data on what RStudio appears to be doing during these hangs (e.g. via Process Explorer, or other similar Windows [Sysinternals](https://docs.microsoft.com/en-us/sysinternals/) tools?)

I'm wondering whether the issue might be something like:

1. The `rsession.exe` process itself is hung, or crashed, due to the aforementioned Boost issue;
2. The `rsession.exe` process is stuck trying to perform Disk IO; e.g. due to latency issues when accessing these network drives.

RStudio does try to write project-specific state to the project folder's (hidden) `.Rproj.user` folder both while the session is running, and during close. I wonder if the attempts to read and write that state are taking a very long time.

Or, if you've configured RStudio to auto-save and auto-load the R workspace, it could also be caused by latency in attempting to serialize / deserialize that workspace (typically a file called `.RData` at the project root).

**Comment by gwd666** [comment]
@kevinushey 
Ouch - that boost update time is really bad news for saving standard projects on network drives. 
I will try to gather as much additional Infos on these delays during project (open, close, switching) transactions. 
I am in a Corp env here and there was a SW update (also to the std OS of the user workstations) shortly before this project switching performance deterioration started showing up.

**Comment by gwd666** [comment]
Screenshot of my General settings [.RData Restore unticked, also deactivated Restore most recent project now - maybe that also helps]
![image](https://user-images.githubusercontent.com/57595487/92694577-8eaab080-f347-11ea-8cfb-aaee1c4194a5.png)

It helps quite a bit if you "Open Projects in a New Session" instead of close/open or directly switching (selecting another project in the dropdown)

**Comment by gwd666** [comment]
> > Final 2 questions when do u expect yr update and will this boost update also effect RStudio 1.2.5033? Thanks
> 
> Sorry, but I'm not sure -- most likely, this update won't happen until next year; perhaps with the next version of RStudio (v1.5). I'm realize this isn't what you wanted to hear, but updating dependencies is time-consuming as it takes time for the team to run down regressions and other issues that typically arise after an update.
> 
> Per your most recent comment -- this sounds extraordinarily frustrating; I'm sorry that RStudio has been causing you so much trouble here. :-/ Are you able to collect any diagnostic data on what RStudio appears to be doing during these hangs (e.g. via Process Explorer, or other similar Windows [Sysinternals](https://docs.microsoft.com/en-us/sysinternals/) tools?)
> 
> I'm wondering whether the issue might be something like:
> 
> 1. The `rsession.exe` process itself is hung, or crashed, due to the aforementioned Boost issue;
> 2. The `rsession.exe` process is stuck trying to perform Disk IO; e.g. due to latency issues when accessing these network drives.
> 
> RStudio does try to write project-specific state to the project folder's (hidden) `.Rproj.user` folder both while the session is running, and during close. I wonder if the attempts to read and write that state are taking a very long time.
> 
> Or, if you've configured RStudio to auto-save and auto-load the R workspace, it could also be caused by latency in attempting to serialize / deserialize that workspace (typically a file called `.RData` at the project root).

Tha setting is deactivated. 
I have come to realize that RStudio does not totally "hang up" - I was usually just to anxious and killing the processes after looking at blank or white startup screens for over 5 minutes (usually not even that clocky circle thing rotating in the middle) and not detecting any progress. Only after I took the time (ie wait for up to 10 minutes depending on the actions taken) I have come to realize that it does not crash BUT just takes vast amounts of time to complete the tasks - for some I have already found some workarounds (eg always open the next project as an "additional" one in a New Session, change some Settings - see other comment).

**Comment by kevinushey** [comment]
It may also be worth checking whether these projects are configured specifically to save + restore the R workspace, via `Tools -> Project Options...` in that open project:

<img width="557" alt="Screen Shot 2020-09-10 at 11 13 58 AM" src="https://user-images.githubusercontent.com/1976582/92779243-bb19f900-f356-11ea-8b40-588803605ec3.png">

I am a bit surprised to see Robocopy here, though -- RStudio doesn't explicitly use it anywhere when attempting to write or copy files. Is it possible that an R package loaded in the session is trying to use `robocopy` to copy files? Or is there another reason why it might be in use?

The `renv` package does use `robocopy` when attempting to copy directories on Windows; are you by any chance using `renv` in these projects?

**Comment by gwd999** [comment]
> It may also be worth checking whether these projects are configured specifically to save + restore the R workspace, via `Tools -> Project Options...` in that open project:
> 
> <img alt="Screen Shot 2020-09-10 at 11 13 58 AM" width="557" src="https://user-images.githubusercontent.com/1976582/92779243-bb19f900-f356-11ea-8b40-588803605ec3.png">
> 
> I am a bit surprised to see Robocopy here, though -- RStudio doesn't explicitly use it anywhere when attempting to write or copy files. Is it possible that an R package loaded in the session is trying to use `robocopy` to copy files? Or is there another reason why it might be in use?
> 
> The `renv` package does use `robocopy` when attempting to copy directories on Windows; are you by any chance using `renv` in these projects?

My Project Options -: General are No, No and (Default). 
Yes the slow starting project is using "renv" (from day one ie its inital setup).

**Comment by gwd666** [comment]
> It may also be worth checking whether these projects are configured specifically to save + restore the R workspace, via `Tools -> Project Options...` in that open project:
> 
> <img alt="Screen Shot 2020-09-10 at 11 13 58 AM" width="557" src="https://user-images.githubusercontent.com/1976582/92779243-bb19f900-f356-11ea-8b40-588803605ec3.png">
> 
> I am a bit surprised to see Robocopy here, though -- RStudio doesn't explicitly use it anywhere when attempting to write or copy files. Is it possible that an R package loaded in the session is trying to use `robocopy` to copy files? Or is there another reason why it might be in use?
> 
> The `renv` package does use `robocopy` when attempting to copy directories on Windows; are you by any chance using `renv` in these projects?

image
Project Options -> General are: No, No and (Default) PLUS 'Quit child processes on exit ' is activated (see pic above).
Yes the slow starting project is using "renv" (but does so since day one ie it was already initially set up this way). I could try to deactivate 'renv' and retest behavior, but I would be quite unhappy to loose renv functionality :-(

**Comment by kevinushey** [comment]
I would be unhappy as well!

We could try adjusting some of `renv`'s configuration settings, to see if that helps out with what's going on. What version of `renv` are you using in these projects? Are you able to update to the latest version?

For example, you could set the following R options; e.g. in the project's `.Rprofile` (or even in a running session):

```
options(renv.config.synchronized.check = FALSE)
options(renv.config.copy.method = "r")
```

It might also be worth trying to generate an R profile of what's going on. You can do so with:

```r
utils::Rprof("~/rstudio-trace.Rprof")  # or pick another appropriate path for the trace to be written
```

After running this, try taking the action that is slow on your system (switching projects). After that's done, you can find the file at "~/rstudio-trace.Rprof", and share it with us.

**Comment by gwd666** [comment]
> I would be unhappy as well!
> 
> We could try adjusting some of `renv`'s configuration settings, to see if that helps out with what's going on. What version of `renv` are you using in these projects? Are you able to update to the latest version?
> 
> For example, you could set the following R options; e.g. in the project's `.Rprofile` (or even in a running session):
> 
> ```
> options(renv.config.synchronized.check = FALSE)
> options(renv.config.copy.method = "r")
> ```
> 
> It might also be worth trying to generate an R profile of what's going on. You can do so with:
> 
> ```r
> utils::Rprof("~/rstudio-trace.Rprof")  # or pick another appropriate path for the trace to be written
> ```
> 
> After running this, try taking the action that is slow on your system (switching projects). After that's done, you can find the file at "~/rstudio-trace.Rprof", and share it with us.

@kevinushey -  will give it a shot after the weekend and get back with results

**Comment by gwd666** [comment]
@kevinushey 
Pls find attached the trace file approx 18836 lines of often repeating commands written during switching to the project. 
I copied utils::Rprof("~/rstudio-trace.Rprof") to the projects .Rprofile file, since running it in the Console and then opening the project did not produce anything except 
```
sample.interval=20000
"f" "doTryCatch" "tryCatchOne" "tryCatchList" "tryCatch" "withCallingHandlers" ".rs.callAs" "utils::Rprof" 
```
as output in it. 
Closing the project again also took some time, but only the last 5 lines were written to the trace file during that process ie these ones: 
```
"file.exists" "<Anonymous>" 
"file.exists" "<Anonymous>" 
"file.exists" "<Anonymous>" 
"paste" "<Anonymous>" 
"makeRestartList" "withRestarts" ".signalSimpleWarning" "rm" "withCallingHandlers" "suppressWarnings" "<Anonymous>" 
```

[rstudio-trace.Rprof.log](https://github.com/rstudio/rstudio/files/5230978/rstudio-trace.Rprof.log)

And as a final remark 
Commenting out  the line 
`source("renv/activate.R") `
in the projects ".Rprofile" file also does the trick of normalizing the switching process. 
So it is very probably safe to say that the "renv" package has something to do with that issue?!

G, W

**Comment by kevinushey** [comment]
Thanks! From the trace, it's clear that this is being caused by `renv`. Calling:

```
utils::summaryRprof("rstudio-trace.Rprof.log")
```

shows that most of the time was spent in these functions:

```
$by.total
                                         total.time total.pct self.time
"doTryCatch"                                 375.10     99.55      0.00
"tryCatch"                                   375.10     99.55      0.00
"tryCatchList"                               375.10     99.55      0.00
"tryCatchOne"                                375.10     99.55      0.00
"eval"                                       373.90     99.23      0.08
"withCallingHandlers"                        373.90     99.23      0.00
"eval.parent"                                373.86     99.22      0.00
"local"                                      373.86     99.22      0.00
"source"                                     373.76     99.19      0.10
"withVisible"                                373.36     99.09      0.00
"renv_bootstrap_load"                        373.26     99.06      0.00
"renv::load"                                 373.10     99.02      0.00
"catch"                                      372.86     98.95      0.00
"FUN"                                        372.50     98.86      0.02
"lapply"                                     372.46     98.85      0.02
"renv_load_sandbox"                          372.34     98.82      0.00
"renv_sandbox_activate_impl"                 372.34     98.82      0.00
"renv_sandbox_activate"                      372.34     98.82      0.00
"renv_sandbox_generate"                      372.30     98.81      0.00
"enumerate"                                  371.96     98.72      0.00
"f"                                          371.96     98.72      0.00
"renv_file_link"                             371.94     98.71      0.00
"catchall"                                   369.80     98.14      0.00
"renv_file_copy"                             368.58     97.82      0.00
"renv_file_copy_dir"                         368.18     97.71      0.00
"method"                                     366.52     97.27      0.00
"renv_file_copy_dir_impl"                    366.52     97.27      0.00
"file.copy"                                  359.28     95.35    359.18
```

In particular, it looks like all the time is being spent trying to prepare the `renv` sandbox. Given that, a fix for this would be to set:

```
RENV_CONFIG_SANDBOX_ENABLED = FALSE
```

within an appropriate startup `.Renviron` file. For example, you could place this in the file located at:

```
file.path(R.home("etc"), "Renviron.site")
```

Or, if just for a single user, then at `~/.Renviron`.

**Comment by gwd666** [comment]
@kevinushey thanks a lot - so I guess then I will bring this later issue to the attention of the renv maintainer; the problem with saving standard projects on Windows Network drives still will remain open until the boost update ... 
So I guess after I havin g moved our discussion to the renv repo I will remove/delete most of the otehr topic not related to that. 
Again THANKS for your help! 
Greetings, Walter

**Comment by kevinushey** [comment]
Yes, I'm sure the maintainer of `renv` will be happy to help :-) I'll transfer this issue to that repository.
